src/Bridge/Symfony/Routing/IriConverter.php

```diff
@@ -168,12 +170,21 @@ public function getItemIriFromResourceClass(string $resourceClass, array $identi
     public function getSubresourceIriFromResourceClass(string $resourceClass, array $context, int $referenceType = UrlGeneratorInterface::ABS_PATH): string
     {
         try {
-            return $this->router->generate($this->routeNameResolver->getRouteName($resourceClass, OperationType::SUBRESOURCE, $context), $context['subresource_identifiers'], $referenceType);
+            return $this->generate($this->routeNameResolver->getRouteName($resourceClass, OperationType::SUBRESOURCE, $context), $context['subresource_identifiers'], $referenceType);
         } catch (RoutingExceptionInterface $e) {
             throw new InvalidArgumentException(sprintf('Unable to generate an IRI for "%s".', $resourceClass), $e->getCode(), $e);
         }
     }
 
+    /**
+     * @param array $parameters
+     * @param int   $referenceType
+     */
+    private function generate(string $name, $parameters = [], $referenceType = UrlGeneratorInterface::ABS_PATH): string
```

Comment:
It doesn't look good to me. This override globally a context parameter. A better approach would be to pass `null` as the default value of `$referenceType`, and only if it's null set the good default depending of the global configuration.

Also, please inline this, there is no need to create a new method and it will hurt performance as it's on the hot path.

Reply:
Thanks for this. I agree it was a bit heavy-handed.

Reply:
@dunglas I think I got them all - added tests for `hal`, `jsonld` and `jsonapi` and verified it works for the entrypoints as well.